### DISCUSSION TITLE
prefix in front of CellNames wnen loading separate fragment files

### DISCUSSION THREAD

**Discussion by khain2650** [GreenleafLab/ArchR#1029] [Questions / Documentation] [answered]
Hi all,

I was wondering whether there's a way to disable prefixes in front of cell names? For instance, when using:

inputFiles <- (getInputFiles("E:\\Kimlabdata\\Fragments"))
ArrowFiles <- createArrowFiles(  inputFiles = inputFiles,
                                 sampleNames = names(inputFiles),
                                filterTSS = 4, 
                                filterFrags = 1000, 
                                addTileMat = TRUE,
                               addGeneScoreMat = TRUE)

The cell names in samples I've loaded would look something like control#TGTGATCAGGCTAATC-1 or treated#TGTGATCAGGCTAATC-1. Is there a way to disable these prefixes "control#" "treated#"?

I've tried using: 
        noprefix <- gsub("control#|treated#","",proj$cellNames)
        proj$cellNames <- noprefix

but then I get the error: cellNames is a protected column in cellColData. Please do not try to overwrite this column!

I'm trying to filter out cells using a barcode-containing csv file, and am having a roadblock since the cell names in the csv file doesn't match that in the archRproject due to said prefix. Any help is appreciated!

**Comment by rcorces** [comment, marked as answer]
No - you cannot change cellNames as the error suggests. What about adding a new column to cellColData containing just the barcode info you want?

**Reply by khain2650** [reply]
Thanks for the reply! That's something I've thought about too. Which function or argument would one use to subset cells using the barcode info in this new column (CellNames1)? For instance, the argument cells in subsetArchRProject() would use CellNames; is there a way to specify which column the function uses for subsetting?

**Reply by khain2650** [reply]
Nevermind, figured it out. For future readers who might be interested, I used: 

          neuron.list <- read.csv("E:\\data\\neurons.bc.csv", header = FALSE)$V2
          noprefix <- gsub("control#|treated#","",proj$cellNames)
          proj$noprefix <- noprefix
          subset <- which(proj$noprefix %in% neuron.list)
          subset <- na.omit(subset)
          neurons <- proj[subset, ]
